workflow-service/src/test/java/com/redhat/parodos/project/controller/ProjectControllerTest.java

```diff
@@ -289,11 +298,65 @@ public void testCreateAccessRequestToProjectWhenUserAlreadyAssignedToProject() t
+	public void testGetProjectMembersById() throws Exception {
+		UUID projectId = UUID.randomUUID();
+		ProjectMemberResponseDTO projectMemberResponseDTO1 = createSampleProjectMember(TEST_USERNAME_1,
+				TEST_ROLE_OWNER);
+		ProjectMemberResponseDTO projectMemberResponseDTO2 = createSampleProjectMember(TEST_USERNAME_2, TEST_ROLE_ADMIN,
+				TEST_ROLE_DEVELOPER);
+
+		// When
+		when(projectService.getProjectMembersById(eq(projectId)))
+				.thenReturn(List.of(projectMemberResponseDTO1, projectMemberResponseDTO2));
+
+		mockMvc.perform(this.getRequestWithValidCredentials(String.format("/api/v1/projects/%s/members", projectId)))
+				.andExpect(MockMvcResultMatchers.status().isOk())
+				.andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+				.andExpect(MockMvcResultMatchers.jsonPath("$", Matchers.hasSize(2)))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[0].username", Matchers.is(TEST_USERNAME_1)))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[0].firstName", Matchers.is(TEST_USERNAME_1)))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[0].lastName", Matchers.is(TEST_USERNAME_1)))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[0].roles", Matchers.hasSize(1)))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[0].roles[0]", Matchers.is(TEST_ROLE_OWNER)))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[1].username", Matchers.is(TEST_USERNAME_2)))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[1].firstName", Matchers.is(TEST_USERNAME_2)))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[1].lastName", Matchers.is(TEST_USERNAME_2)))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[1].roles[0]", Matchers.is(TEST_ROLE_ADMIN)))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[1].roles[1]", Matchers.is(TEST_ROLE_DEVELOPER)))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[1].roles", Matchers.hasSize(2)));
+
+		// Then
+		verify(projectService, times(1)).getProjectMembersById(any());
+	}
+
+	@Test
+	public void testGetProjectMembersByIdWithInvalidId() throws Exception {
+		when(projectService.getProjectMembersById(any()))
+				.thenThrow(new ResourceNotFoundException("Project members not found"));
+
+		// When
+		mockMvc.perform(
+				this.getRequestWithValidCredentials(String.format("/api/v1/projects/%s/members", UUID.randomUUID())))
+				.andExpect(MockMvcResultMatchers.status().isNotFound());
+		// Then
+		verify(projectService, times(1)).getProjectMembersById(any());
+	}
+
 	ProjectResponseDTO createSampleProject(String name) {
 		ProjectResponseDTO responseDTO = new ProjectResponseDTO();
 		responseDTO.setId(UUID.randomUUID());
 		responseDTO.setName(name);
 		return responseDTO;
 	}
 
+	ProjectMemberResponseDTO createSampleProjectMember(String name, String... roles) {
+		ProjectMemberResponseDTO responseDTO = new ProjectMemberResponseDTO();
+		responseDTO.setUsername(name);
+		responseDTO.setFirstName(name);
+		responseDTO.setLastName(name);
+		responseDTO.setRoles(new TreeSet<>(Set.of(roles)));
```

Comment:
why don't use `new TreeSet(roles)` (without the `Set.of`)?

Reply:
same reason as above List.of